### PR TITLE
add 201 aci wordpress vnet

### PR DESCRIPTION
**Pull request by tilnl**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: 201-aci-wordpress-vnet/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Create an Azure container group with VNet
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F201-aci-wordpress-vnet%2Fazuredeploy.json" target="_blank">
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F201-aci-wordpress-vnet%2Fazuredeploy.json" target="_blank">
+    <img src="http://armviz.io/visualizebutton.png"/>
+</a>
+
+This template creates a WordPress website and its MySQL database on container Instances in a virtual network. The WordPress site content and MySQL database are persistently stored on an Azure Storage File Share.
```
Comment (wenwu449) [inline 201-aci-wordpress-vnet/README.md]: Maybe describe how is the WordPress exposed to public network?

File: 201-aci-wordpress-vnet/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+## Solution overview and deployed resources
+
+The following resources are deployed as part of the solution
+
++ **Azure Container Instance**: Azure Container Instance to host the WordPress site.
++ **Azure Container Instance**: Azure Container Instance to host the MySQL database.
++ **Azure Container Instance**: A [run-once](https://docs.microsoft.com/en-us/azure/container-instances/container-instances-restart-policy#container-restart-policy) Azure Container Instance, where the az-cli is executed to create the file shares
++ **Storage Account**: Storage account for the file shares to store the WordPress site content and MySQL database
++ **File share**: Azure File shares to store WordPress site content and MySQL database.
++ **Application gateway**: Application gateway for WordPress site.
```
Comment (wenwu449) [inline 201-aci-wordpress-vnet/README.md]: does the gateway expose public network access to wordpress in vnet?

File: 201-aci-wordpress-vnet/metadata.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+  "type": "QuickStart",
+  "itemDisplayName": "Create a WordPress site",
```
Comment (wenwu449) [inline 201-aci-wordpress-vnet/metadata.json]: mention "vnet"?

File: 201-aci-wordpress-vnet/metadata.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  "description": "This template creates a WordPress site on Container Instances in a virtual network.",
```
Comment (wenwu449) [inline 201-aci-wordpress-vnet/metadata.json]: mention public internet accesss?

File: 201-aci-wordpress-vnet/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,484 @@
...
+      "apiVersion": "2017-10-01",
+      "location": "[parameters('location')]",
+      "sku": {
+        "name": "[variables('storageAccountType')]"
+      },
+      "kind": "Storage",
+      "properties": { }
+    },
+    {
+      "apiVersion": "2015-05-01-preview",
```
Comment (bmoore-msft) [inline 201-aci-wordpress-vnet/azuredeploy.json]: update apiVersion

File: 201-aci-wordpress-vnet/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,484 @@
...
+      "type": "Microsoft.Network/publicIPAddresses",
+      "name": "[variables('publicIPAddressName')]",
+      "location": "[parameters('location')]",
+      "properties": {
+        "publicIPAllocationMethod": "Dynamic",
+         "dnsSettings": {
+          "domainNameLabel": "[variables('siteName')]"
```
Comment (bmoore-msft) [inline 201-aci-wordpress-vnet/azuredeploy.json]: this needs to be globally unique - you'll need to generate something with uniqueString()

File: 201-aci-wordpress-vnet/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,484 @@
...
+        ]
+      }
+    },
+    {
+      "name": "[variables('shareContainerGroupName')]",
+      "type": "Microsoft.ContainerInstance/containerGroups",
+      "apiVersion": "2018-02-01-preview",
+      "location": "[parameters('location')]",
+      "dependsOn": [
+        "[concat('Microsoft.Storage/storageAccounts/', variables('storageAccountName'))]"
```
Comment (bmoore-msft) [inline 201-aci-wordpress-vnet/azuredeploy.json]: remove concat and use resourceId()

File: 201-aci-wordpress-vnet/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,484 @@
...
+        "osType": "Linux"
+      }
+    },
+    {
+      "name": "[variables('mysqlContainerGroupName')]",
+      "type": "Microsoft.ContainerInstance/containerGroups",
+      "apiVersion": "2018-07-01",
+      "location": "[parameters('location')]",
+      "dependsOn": [
+        "[concat('Microsoft.ContainerInstance/containerGroups/', variables('shareContainerGroupName'))]",
```
Comment (bmoore-msft) [inline 201-aci-wordpress-vnet/azuredeploy.json]: remove concat and use resourceId()

File: 201-aci-wordpress-vnet/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,484 @@
...
+        "[concat('Microsoft.Network/networkProfiles/', variables('networkProfileName'))]"
```
Comment (bmoore-msft) [inline 201-aci-wordpress-vnet/azuredeploy.json]: remove concat and use resourceId()

File: 201-aci-wordpress-vnet/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,484 @@
...
+        "osType": "Linux"
+      }
+    },
+    {
+      "name": "[variables('wordpressContainerGroupName')]",
+      "type": "Microsoft.ContainerInstance/containerGroups",
+      "apiVersion": "2018-07-01",
+      "location": "[parameters('location')]",
+      "dependsOn": [
+        "[concat('Microsoft.ContainerInstance/containerGroups/', variables('shareContainerGroupName'))]",
```
Comment (bmoore-msft) [inline 201-aci-wordpress-vnet/azuredeploy.json]: remove concat and use resourceId()

File: 201-aci-wordpress-vnet/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,484 @@
...
+        "[concat('Microsoft.ContainerInstance/containerGroups/', variables('mysqlContainerGroupName'))]"
```
Comment (bmoore-msft) [inline 201-aci-wordpress-vnet/azuredeploy.json]: remove concat and use resourceId()

File: 201-aci-wordpress-vnet/azuredeploy.json [hunk 12]
```diff
@@ -0,0 +1,484 @@
...
+        "osType": "Linux"
+      }
+    },
+    {
+      "apiVersion": "2017-06-01",
+      "name": "[variables('applicationGatewayName')]",
+      "type": "Microsoft.Network/applicationGateways",
+      "location": "[parameters('location')]",
+      "dependsOn": [
+        "[concat('Microsoft.Network/virtualNetworks/', parameters('vnetName'))]",
```
Comment (bmoore-msft) [inline 201-aci-wordpress-vnet/azuredeploy.json]: remove concat and use resourceId()

File: 201-aci-wordpress-vnet/azuredeploy.json [hunk 13]
```diff
@@ -0,0 +1,484 @@
...
+        "[concat('Microsoft.Network/publicIPAddresses/', variables('publicIPAddressName'))]",
```
Comment (bmoore-msft) [inline 201-aci-wordpress-vnet/azuredeploy.json]: remove concat and use resourceId()

File: 201-aci-wordpress-vnet/azuredeploy.json [hunk 14]
```diff
@@ -0,0 +1,484 @@
...
+        "[concat('Microsoft.ContainerInstance/containerGroups/', variables('wordpressContainerGroupName'))]"
```
Comment (bmoore-msft) [inline 201-aci-wordpress-vnet/azuredeploy.json]: remove concat and use resourceId()

File: 201-aci-wordpress-vnet/azuredeploy.json [hunk 15]
```diff
@@ -0,0 +1,484 @@
...
+              "CookieBasedAffinity": "Disabled"
+            }
+          }
+        ],
+        "httpListeners": [
+          {
+            "name": "appGatewayHttpListener",
+            "properties": {
+              "FrontendIPConfiguration": {
+                "Id": "[concat(variables('applicationGatewayID'), '/frontendIPConfigurations/appGatewayFrontendIP')]"
```
Comment (bmoore-msft) [inline 201-aci-wordpress-vnet/azuredeploy.json]: remove concat and use resourceId()

File: 201-aci-wordpress-vnet/azuredeploy.json [hunk 16]
```diff
@@ -0,0 +1,484 @@
...
+              },
+              "FrontendPort": {
+                "Id": "[concat(variables('applicationGatewayID'), '/frontendPorts/appGatewayFrontendPort')]"
```
Comment (bmoore-msft) [inline 201-aci-wordpress-vnet/azuredeploy.json]: remove concat and use resourceId()

File: 201-aci-wordpress-vnet/azuredeploy.json [hunk 17]
```diff
@@ -0,0 +1,484 @@
...
+            }
+          }
+        ],
+        "requestRoutingRules": [
+          {
+            "Name": "rule1",
+            "properties": {
+              "RuleType": "Basic",
+              "httpListener": {
+                "id": "[concat(variables('applicationGatewayID'), '/httpListeners/appGatewayHttpListener')]"
```
Comment (bmoore-msft) [inline 201-aci-wordpress-vnet/azuredeploy.json]: remove concat and use resourceId()

File: 201-aci-wordpress-vnet/azuredeploy.json [hunk 18]
```diff
@@ -0,0 +1,484 @@
...
+              },
+              "backendAddressPool": {
+                "id": "[concat(variables('applicationGatewayID'), '/backendAddressPools/appGatewayBackendPool')]"
```
Comment (bmoore-msft) [inline 201-aci-wordpress-vnet/azuredeploy.json]: remove concat and use resourceId()

File: 201-aci-wordpress-vnet/azuredeploy.json [hunk 19]
```diff
@@ -0,0 +1,484 @@
...
+              },
+              "backendHttpSettings": {
+                "id": "[concat(variables('applicationGatewayID'), '/backendHttpSettingsCollection/appGatewayBackendHttpSettings')]"
```
Comment (bmoore-msft) [inline 201-aci-wordpress-vnet/azuredeploy.json]: remove concat and use resourceId()

**Pull request merged**